modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/index.d.ts

```diff
@@ -174,13 +139,49 @@ interface ObjectDefinition {
 	titleObjectFieldName: string;
 }
 
+interface SelectOption {
+	label: string;
+	value: string;
+}
+
+type APIApplicationUIData = Pick<
+	APIApplicationItem,
+	'baseURL' | 'description' | 'title'
+>;
+
+type APIEndpointUIData = Pick<
+	APIEndpointItem,
+	'description' | 'path' | 'scope'
+>;
+
+type APISchemaUIData = Pick<
+	APISchemaItem,
+	'description' | 'name' | 'mainObjectDefinitionERC'
+>;
+
+type ActiveNav = 'details' | 'endpoints' | 'schemas';
+
+type ApplicationDataError = {
+	baseURL: boolean;
+	title: boolean;
+};
+
+type ApplicationStatusKeys = 'published' | 'unpublished';
+
+type EndpointDataError = {
+	description: boolean;
+	path: boolean;
+	scope: boolean;
+};
+
+type MainEndpointNav = 'list' | {edit: number};
```

Comment:
@lhine I just noticed this is interface is the same as the one defined below. Could you please delete one of them and rename the remaining one with a more generic name? Thanks!